server/src/test/java/org/elasticsearch/action/admin/cluster/allocation/TransportGetAllocationStatsActionTests.java

```diff
@@ -230,33 +274,47 @@ public void testGetStatsWithCachingEnabled() throws Exception {
         // Initial cache miss, all threads should get the same value.
         resetExpectedAllocationStats.run();
         ESTestCase.startInParallel(between(1, 5), threadNumber -> safeAwait(threadTask));
-        verify(allocationStatsService, times(++numExpectedAllocationStatsServiceCalls)).stats();
+        verifyAllocationStatsServiceNumCallsEqualTo(++numExpectedAllocationStatsServiceCalls);
 
         // Advance the clock to a time less than or equal to the TTL and verify we still get the cached stats.
         threadPool.setCurrentTimeInMillis(startTimeMillis + between(0, (int) allocationStatsCacheTTL.millis()));
         ESTestCase.startInParallel(between(1, 5), threadNumber -> safeAwait(threadTask));
-        verify(allocationStatsService, times(numExpectedAllocationStatsServiceCalls)).stats();
+        verifyAllocationStatsServiceNumCallsEqualTo(numExpectedAllocationStatsServiceCalls);
 
         // Force the cached stats to expire.
         threadPool.setCurrentTimeInMillis(startTimeMillis + allocationStatsCacheTTL.getMillis() + 1);
 
         // Expect a single call to the stats service on the cache miss.
         resetExpectedAllocationStats.run();
         ESTestCase.startInParallel(between(1, 5), threadNumber -> safeAwait(threadTask));
-        verify(allocationStatsService, times(++numExpectedAllocationStatsServiceCalls)).stats();
+        verifyAllocationStatsServiceNumCallsEqualTo(++numExpectedAllocationStatsServiceCalls);
 
         // Update the TTL setting to disable the cache, we expect a service call each time.
         setAllocationStatsCacheTTL(TimeValue.ZERO);
         safeAwait(threadTask);
         safeAwait(threadTask);
         numExpectedAllocationStatsServiceCalls += 2;
-        verify(allocationStatsService, times(numExpectedAllocationStatsServiceCalls)).stats();
+        verifyAllocationStatsServiceNumCallsEqualTo(numExpectedAllocationStatsServiceCalls);
 
         // Re-enable the cache, only one thread should call the stats service.
-        setAllocationStatsCacheTTL(TimeValue.timeValueMinutes(5));
+        final var newTTL = TimeValue.timeValueMinutes(5);
+        setAllocationStatsCacheTTL(newTTL);
+        threadPool.setCurrentTimeInMillis(threadPool.relativeTimeInMillis() + newTTL.getMillis() + 1);
         resetExpectedAllocationStats.run();
         ESTestCase.startInParallel(between(1, 5), threadNumber -> safeAwait(threadTask));
-        verify(allocationStatsService, times(++numExpectedAllocationStatsServiceCalls)).stats();
+        verifyAllocationStatsServiceNumCallsEqualTo(++numExpectedAllocationStatsServiceCalls);
+    }
+
+    private void verifyAllocationStatsServiceNumCallsEqualTo(int numCalls) {
+        verify(allocationStatsService, times(numCalls)).stats(argThat(r -> true));
```

Comment:
Is this the same thing?

```suggestion
        verify(allocationStatsService, times(numCalls)).stats(any());
```

Reply:
Yes, I applied this in the other uses as well.